server/controllers/events.js

```diff
@@ -35,4 +35,28 @@ module.exports = {
       console.log(err);
     }
   },
+  deleteEvent: async (req, res) => {
+    try {
+      const eventId = req.params.id;
+      //checks if an event exists that _id, user, and req.user._id match. This is to prevent users that are authenticated from deleting events they do not author.
+      const event = await Event.findOne({ _id: eventId, user: req.user._id });
+      if (!event) { return res.status(401).send({ message: 'You are not the author of this event' }); }
+      await Event.deleteOne({ _id: eventId });
+      res.json({ message: 'Event deleted' });
+    } catch (error) {
+      console.error(error);
+    }
```

Comment:
a non-blocking comment:

In general, we want to make try/catch as surgical as possible - this kind of precision allows for us to know what failed on the server (e.g. did we fail on the `findOne` call, or fail on the `deleteOne` call?)  In this case it's not _as_ important since neither call is really prone to erroring, but doing blanket try/catches are considered not good practice.  Errors are your friends! :) They tell you where your code is breaking so you can fix it accordingly.

Additionally, when we catch an error, we should always respond to the client.  In this case, it would be enough to do something like:
```js
res.send(500);
```
after we log the error to the console.  That way the client knows that it failed and can handle it accordingly.

Reply:
> a non-blocking comment:
> 
> In general, we want to make try/catch as surgical as possible - this kind of precision allows for us to know what failed on the server (e.g. did we fail on the `findOne` call, or fail on the `deleteOne` call?) In this case it's not _as_ important since neither call is really prone to erroring, but doing blanket try/catches are considered not good practice. Errors are your friends! :) They tell you where your code is breaking so you can fix it accordingly.
> 
> Additionally, when we catch an error, we should always respond to the client. In this case, it would be enough to do something like:
> 
> ```js
> res.send(500);
> ```
> 
> after we log the error to the console. That way the client knows that it failed and can handle it accordingly.

That made sense. Any idea why the .send({ message: 'You are not the author of this event' }); was not being received even though I was receiving a 401 error?

Reply:
I think you meant to do `.json()` instead